packages/jsapi-utils/src/ConnectionUtils.test.ts

```diff
@@ -0,0 +1,114 @@
+import dh, { VariableDefinition } from '@deephaven/jsapi-shim';
+import { TimeoutError } from '@deephaven/utils';
+import { fetchVariableDefinition } from './ConnectionUtils';
+
+const mockRemoveListener = jest.fn();
+const mockSubscribeToFieldUpdates = jest.fn(
+  changeListener => mockRemoveListener
+);
+
+const connection = new dh.IdeConnection('http://mockserver');
+connection.subscribeToFieldUpdates = mockSubscribeToFieldUpdates;
+
+const testDefinition1: VariableDefinition = {
+  title: 'TEST_DEF',
+  type: dh.VariableType.TABLE,
+};
+
+const testDefinition2: VariableDefinition = {
+  title: 'ANOTHER_DEF',
+  type: dh.VariableType.FIGURE,
+};
+
+beforeAll(() => {
+  jest.useFakeTimers();
+});
+
+afterAll(() => {
+  jest.useRealTimers();
+});
+
+beforeEach(() => {
+  mockRemoveListener.mockClear();
+  mockSubscribeToFieldUpdates.mockClear();
+});
+
+it('finds the right definition if variable exists', async () => {
+  const fetchPromise = fetchVariableDefinition(
+    connection,
+    testDefinition1.title ?? ''
```

Comment:
Why `?? ''`?

Reply:
This would be a case where the `satisfies` operator would be handy.
`testDefinition1` is defined as a `VariableDefintion`, where `title` is optional; even though I very clearly define it prior to this. I'll rewrite this some other way to be cleaner.

Reply:
You just don't need to define the type on `testDefinition1` and `testDefinition2`. TS will figure it out anywhere that it needs a `VariableDefinition`. If in the future `VariableDefinition` changes, anywhere that requires it within the test will complain since the variables won't match the updated type

Reply:
Yea I just don't need to define the type. I think I did originally for autocomplete. D'oh